src/main/java/net/neoforged/neoforge/common/util/InsertableLinkedOpenCustomHashSet.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.common.util;
+
+import it.unimi.dsi.fastutil.Hash;
+import it.unimi.dsi.fastutil.HashCommon;
+import it.unimi.dsi.fastutil.objects.ObjectLinkedOpenCustomHashSet;
+import java.util.Objects;
+import net.neoforged.neoforge.common.util.strategy.BasicStrategy;
+import net.neoforged.neoforge.common.util.strategy.IdentityStrategy;
+
+/**
+ * Special linked hash set that allow changing the order of its entries and is strict to throw if attempting to add an entry that already exists.
+ * Requires a strategy for the hashing behavior. Use {@link InsertableLinkedOpenCustomHashSet#BASIC} or {@link InsertableLinkedOpenCustomHashSet#IDENTITY} if no special hashing needed.
+ */
+public class InsertableLinkedOpenCustomHashSet<T> extends ObjectLinkedOpenCustomHashSet<T> {
+    /**
+     * A strategy that uses {@link Objects#hashCode(Object)} and {@link Object#equals(Object)}.
+     */
+    public static final Hash.Strategy<? super Object> BASIC = new BasicStrategy();
+
+    /**
+     * A strategy that uses {@link System#identityHashCode(Object)} and {@code a == b} comparisons.
+     */
+    public static final Hash.Strategy<? super Object> IDENTITY = new IdentityStrategy();
```

Comment:
Would it be reasonable to make these static instances in their respective classes?

Reply:
For uh some reason(tm) I thought the strategy was a fast util thing... lol